### ISSUE TITLE
Hacks shortcuts not detecting FCPX even though it's installed

### ISSUE THREAD

**Issue by latenitefilms** [fcpxhacks/fcpxhacks#702]
@randomeizer - When you re-wrote the Hacks Shortcut code, did you allow for situations where users only have their language installed (i.e. they've used "Manage Storage" to remove languages other than the one they use)?

Claudio reports via the Feedback Form:

> What went wrong?
> I tried to enable "Control CommandPost Shortcuts within Final Cut Pro"
> 
> What did you expect to happen?
> to control the shortcuts within FCPXâ€¦ faster than throughout the command post
> 
> What are the steps to recreate the problem?
> Preference->Final Cut pro->Control CommandPost Shortcuts within Final Cut Pro
> 
> And i received the message: 
> No supported versions of Final Cut Pro were detected.

Error Log:

```
CommandPost v1.0.0-beta.9

macOS Version:                  10.12.5
CommandPost Locale:             en
Current Keyboard Layout:        U.S.
Final Cut Pro Path:             /Applications/Final Cut Pro.app
Final Cut Pro Version:          10.3.4
Final Cut Pro Language:         en_IT
Developer Mode:                 false

2017-06-25 14:48:46: 2017-06-25 14:48:44: -- Loading /Users/claudio/Applications/CommandPost.app/Contents/Resources/extensions/cp/init.lua
2017-06-25 14:48:44: -- Loading extension: uielement
2017-06-25 14:48:44: -- Loading extension: window
2017-06-25 14:48:45: ********
2017-06-25 14:48:45: 14:48:45 ERROR:   menubar: Unable to load MainMenu.nib for specified language: de
2017-06-25 14:48:45: ********
2017-06-25 14:48:45: ********
2017-06-25 14:48:45: 14:48:45 ERROR:   menubar: Unable to load MainMenu.nib for specified language: fr
2017-06-25 14:48:45: ********
2017-06-25 14:48:45: ********
2017-06-25 14:48:45: 14:48:45 ERROR:   menubar: Unable to load MainMenu.nib for specified language: ja
2017-06-25 14:48:45: ********
2017-06-25 14:48:45: ********
2017-06-25 14:48:45: 14:48:45 ERROR:   menubar: Unable to load MainMenu.nib for specified language: zh_CN
2017-06-25 14:48:45: ********
2017-06-25 14:48:46: 14:48:46         cp:     Loading Plugins...
2017-06-25 14:48:47: 14:48:47                 Plugins Loaded.
2017-06-25 14:48:48: -- Done.
2017-06-25 14:49:03: 14:49:03 ** Warning:   menubar: Searching manually for 'Final Cut Pro' in ''.
2017-06-25 14:49:03: 14:49:03 ** Warning:   strings: Unable to find 'CommandSubmenu' in 'en_IT'
2017-06-25 14:49:03: 14:49:03 ** Warning:   menubar: Searching manually for 'Commands' in 'Final Cut Pro'.
2017-06-25 14:49:03: 14:49:03 ** Warning:   strings: Unable to find 'Customize' in 'en_IT'
2017-06-25 14:49:03: 14:49:03 ** Warning:   menubar: Searching manually for 'Customize?' in 'Final Cut Pro, Commands'.
2017-06-25 14:49:03: 14:49:03 ** Warning:   menubar: Searching manually for 'Final Cut Pro' in ''.
2017-06-25 14:49:03: 14:49:03 ** Warning:   strings: Unable to find 'CommandSubmenu' in 'en_IT'
2017-06-25 14:49:03: 14:49:03 ** Warning:   menubar: Searching manually for 'Commands' in 'Final Cut Pro'.
2017-06-25 14:49:03: 14:49:03 ** Warning:   strings: Unable to find 'Customize' in 'en_IT'
2017-06-25 14:49:03: 14:49:03 ** Warning:   menubar: Searching manually for 'Customize?' in 'Final Cut Pro, Commands'.
2017-06-25 14:52:32: 14:52:32 ** Warning:   menubar: Searching manually for 'Final Cut Pro' in ''.
2017-06-25 14:52:32: 14:52:32 ** Warning:   strings: Unable to find 'CommandSubmenu' in 'en_IT'
2017-06-25 14:52:32: 14:52:32 ** Warning:   menubar: Searching manually for 'Commands' in 'Final Cut Pro'.
2017-06-25 14:52:32: 14:52:32 ** Warning:   strings: Unable to find 'Customize' in 'en_IT'
2017-06-25 14:52:32: 14:52:32 ** Warning:   menubar: Searching manually for 'Customize?' in 'Final Cut Pro, Commands'.
2017-06-25 14:52:32: 14:52:32 ** Warning:   menubar: Searching manually for 'Final Cut Pro' in ''.
2017-06-25 14:52:32: 14:52:32 ** Warning:   strings: Unable to find 'CommandSubmenu' in 'en_IT'
2017-06-25 14:52:32: 14:52:32 ** Warning:   menubar: Searching manually for 'Commands' in 'Final Cut Pro'.
2017-06-25 14:52:32: 14:52:32 ** Warning:   strings: Unable to find 'Customize' in 'en_IT'
2017-06-25 14:52:32: 14:52:32 ** Warning:   menubar: Searching manually for 'Customize?' in 'Final Cut Pro, Commands'.
2017-06-25 21:54:27: ********
2017-06-25 21:54:27: 21:54:27 ERROR:   wfilter: app SystemUIServer is not registered!
2017-06-25 21:54:27: ********
2017-06-25 21:57:05: ********
2017-06-25 21:57:05: 21:57:05 ERROR:   wfilter: app SystemUIServer is not registered!
2017-06-25 21:57:05: ********
2017-06-25 22:07:16: ********
2017-06-25 22:07:16: 22:07:16 ERROR:   wfilter: app SystemUIServer is not registered!
2017-06-25 22:07:16: ********
2017-06-25 23:12:24: ********
2017-06-25 23:12:24: 23:12:24 ERROR:   LuaSkin: hs.hotkey callback error: ...t.app/Contents/Resources/extensions/cp/ui/ScrollArea.lua:167: attempt to index a nil value (global 'vScroll')
stack traceback:
	...t.app/Contents/Resources/extensions/cp/ui/ScrollArea.lua:167: in function 'cp.ui.ScrollArea.showChild'
	.../extensions/cp/apple/finalcutpro/main/EffectsBrowser.lua:231: in function 'cp.apple.finalcutpro.main.EffectsBrowser.applyItem'
	...tents/Resources/plugins/finalcutpro/timeline/effects.lua:257: in field 'apply'
	...tents/Resources/plugins/finalcutpro/timeline/effects.lua:545: in function <...tents/Resources/plugins/finalcutpro/timeline/effects.lua:545>
	(...tail calls...)
2017-06-25 23:12:24: ********
2017-06-25 23:12:45: ********
2017-06-25 23:12:45: 23:12:45 ERROR:   LuaSkin: hs.hotkey callback error: ...t.app/Contents/Resources/extensions/cp/ui/ScrollArea.lua:167: attempt to index a nil value (global 'vScroll')
stack traceback:
	...t.app/Contents/Resources/extensions/cp/ui/ScrollArea.lua:167: in function 'cp.ui.ScrollArea.showChild'
	.../extensions/cp/apple/finalcutpro/main/EffectsBrowser.lua:231: in function 'cp.apple.finalcutpro.main.EffectsBrowser.applyItem'
	...tents/Resources/plugins/finalcutpro/timeline/effects.lua:257: in field 'apply'
	...tents/Resources/plugins/finalcutpro/timeline/effects.lua:545: in function <...tents/Resources/plugins/finalcutpro/timeline/effects.lua:545>
	(...tail calls...)
2017-06-25 23:12:45: ********
```

**Comment by latenitefilms** [comment]
I'm not sure why the error log is reporting `Final Cut Pro Language: en_IT` - as this should only return one of the "supported" languages? Something must have changed in the `require("cp.apple.finalcutpro"):getCurrentLanguage()` code too?

**Comment by randomeizer** [comment]
Yeah, it's probably not handling it - I wasn't aware you could remove them.

And yes, I modified that code, so it would currently be returning the full
language code for the system in those situations.

**Comment by randomeizer** [comment]
Should be fixed in #667
